src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqModellingGroupRepository.kt

```diff
@@ -142,9 +141,59 @@ class JooqModellingGroupRepository(
         ), scenarioAndData.tableData)
     }
 
-    private fun convertScenarioToResponsibility(scenario: Scenario): Responsibility
+    private fun convertScenarioToResponsibility(scenario: Scenario, responsibilityId: Int): Responsibility
     {
-        return Responsibility(scenario, ResponsibilityStatus.EMPTY, emptyList(), null)
+        val burdenEstimate = getCurrentBurdenEstimate(responsibilityId)
+        val problems: MutableList<String> = mutableListOf()
+        val status = if (burdenEstimate == null)
+        {
+            ResponsibilityStatus.EMPTY
+        }
+        else
+        {
+            if (burdenEstimate.problems.any())
+            {
+                problems.addAll(burdenEstimate.problems)
+                ResponsibilityStatus.INVALID
+            }
+            else
+            {
+                ResponsibilityStatus.VALID
+            }
+        }
+        return Responsibility(scenario, status, problems, burdenEstimate)
+    }
+
+    private fun getCurrentBurdenEstimate(responsibilityId: Int): BurdenEstimateSet?
```

Comment:
this should really be a public function on the burden estimate repo, just waiting on i121 so that that repo exists!

Reply:
Cool, we can move that across in a bit. In the mean time, could you rename `burdenEstimate` to `burdenEstimateSet` throughout this section?